elide-datastore/elide-datastore-hibernate5/src/main/java/com/yahoo/elide/datastores/hibernate5/HibernateTransaction.java

```diff
@@ -207,6 +252,71 @@ public void commit() {
     }
 
     @Override
+    public <T> Collection filterSortOrPaginateCollection(final Collection collection, final Class<T> entityClass,
+                                                         final EntityDictionary dictionary,
+                                                         final Optional<Set<Predicate>> filters,
+                                                         final Optional<Sorting> sorting,
+                                                         final Optional<Pagination> pagination) {
+        if (((collection instanceof AbstractPersistentCollection))
+                && (filters.isPresent() || sorting.isPresent() || pagination.isPresent())) {
+
+            String filterString = "";
+
+            // apply filtering - eg where clause's
+            if (filters.isPresent()) {
+                filterString += hqlFilterOperation.applyAll(filters.get());
```

Comment:
This might have been a bug already in Elide, but we have a global hqlFilterOperation per transaction.  PersistentResource.getRelation will be called multiple times for different collections.  Filters will keep getting added to it (that really don't apply).    I think we should create a new hqlFilterOperation for each filter operation.